cloudclient/options.go

```diff
@@ -132,6 +140,14 @@ func (o *Options) compute() (
 	if version == "" {
 		version = defaultAPIVersion
 	}
+
+	// setup user-agent string
+	userAgent := fmt.Sprintf("temporalio-cloud-sdk-go/%s", sdkVersion)
+	if o.UserAgent != "" {
+		userAgent = fmt.Sprintf("%s %s", strings.TrimSpace(o.UserAgent), userAgent)
+	}
```

Comment:
I think we should consider using `client-name` and `client-version` headers/metadata like all other SDKs